packages/melos/lib/src/common/utils.dart

```diff
@@ -213,25 +213,19 @@ Future<int> startProcess(
     return _arg;
   }).where((element) => element != null);
 
+  final scriptSource = filteredArgs.join(' ');
   final execProcess = await Process.start(
     executable,
-    currentPlatform.isWindows ? ['/C', '%MELOS_SCRIPT%'] : [],
+    currentPlatform.isWindows ? ['/C', '%MELOS_SCRIPT%'] : ['-c', scriptSource],
```

Comment:
```suggestion
    currentPlatform.isWindows ? ['/C', '%MELOS_SCRIPT%'] : ['-c', r'$MELOS_SCRIPT'],
```

Wouldn't this work? Keeps it consistent with how Windows does it too then

Reply:
It causes wrong variable expansion, you can see in test:
`bin/sh: 1: "dart: not found`

single argument `"dart analyze . --fatal-infos"`
becomes `"dart` `analyze` `.` `--fatal-infos"`

Reply:
@Salakar ^

Reply:
I see now, will revert, one moment

Reply:
What's your thoughts on `['-c', r'eval "$MELOS_SCRIPT"']` - since this does work

Reply:
Have tested on Melos and FlutterFire and everything seems to be working with the above so will merge - I can change in another PR if needed. Thanks for your help :)